src/persist-txn/src/txn_cache.rs

```diff
@@ -133,15 +132,13 @@ pub struct TxnsCache<T: Timestamp + Lattice + Codec64, C: TxnsCodec = TxnsCodecD
 }
 
 impl<T: Timestamp + Lattice + TotalOrder + StepForward + Codec64> TxnsCacheState<T> {
-    /// Creates a new [`TxnsCacheState`] that is valid as of the beginning of
-    /// time.
+    /// Creates a new empty [`TxnsCacheState`].
     ///
-    /// All callers should be using [`TxnsCacheState::init`] instead, since the
-    /// actual txn shard is likely compacted past the beginning of time.
-    fn new(txns_id: ShardId, only_data_id: Option<ShardId>) -> Self {
+    /// `init_ts` must be >= the since of the txn shard.
```

Comment:
Yes, you're right it must be == to the critical handle's since. Any timestamp > the critical handle's since may not have been applied, but we make assumptions that everything < init_ts has been applied.

Is `ReadHandle::since()` the right way to get this value?

Reply:
I think it can technically be <= the critical handle's since (which is what I meant to write here) since the important invariant is maintained (that every entry < init_ts is applied).